src/components/Modals/BaseModal/BaseModal.tsx

```diff
@@ -0,0 +1,100 @@
+/* External dependencies */
+import React, {
+  forwardRef,
+  useCallback,
+  useEffect,
+  useRef,
+} from 'react'
+import ReactDom from 'react-dom'
+
+/* Internal dependencies */
+import { document, getRootElement } from 'Utils/domUtils'
+import * as Styled from './BaseModal.styled'
+import { BaseModalProps } from './BaseModal.types'
+
+function BaseModal(
+  {
+    children,
+    show,
+    onHide,
+    autoFocus = true,
+    padded = true,
+    backdropClassName = '',
+    targetElement = getRootElement(),
+    zIndex = 1e7,
+    style,
+    ...rests
+  }: BaseModalProps,
+  forwardedRef: React.Ref<HTMLDivElement>,
+) {
+  const baseModalBodyRef = useRef<HTMLDivElement>(null)
+
+  const onClickBackgroundOverlay = useCallback((e: React.MouseEvent<HTMLDivElement>) => {
+    e.stopPropagation()
+    onHide()
+  }, [onHide])
+
+  const onCloseModalByEsc = useCallback((e: KeyboardEvent) => {
```

Comment:
```suggestion
  const handleKeydown = useCallback((e: KeyboardEvent) => {
```
handleKeydown이 더 적합해보입니다!

Reply:
여기는 onKeydown을 handle하지는 않습니다. esc만 handle합니다.